stats_overlay.js

```diff
@@ -0,0 +1,394 @@
+/**
+ * Copyright 2014 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @fileoverview A statistics visualiser.
+ */
+
+ goog.provide('shaka.StatsOverlay');
+
+/** @constructor
+ * @export
+*/
+shaka.StatsOverlay = function() {
+
+  /** @private {?shaka.player.Player} */
+  this.player_;
+
+  /** @private {?HTMLElement} */
+  this.canvas_;
+
+  /** @private {?HTMLVideoElement} */
+  this.video_;
+
+  /** @private {?number} */
+  this.width_;
+
+  /** @private {?number} */
+  this.height_;
+
+  /** @private {number} */
+  this.maxY_ = 1;
+
+  /** @private {number} */
+  this.maxBufferLength_ = 1;
+
+  /** @private {?number} */
+  this.startTimestamp_;
+
+  /** @private {?number} */
+  this.currentXOffset_;
+
+  /** @private {?number} */
+  this.refreshIntervalId_;
+
+  /** @private {?Array} */
+  this.bufferLengthHistory_;
+
+};
+
+
+/**
+ * Initializes the stats ovelay.
+ * @param {!shaka.player.Player} player
+ * @param {?HTMLElement} canvas
+*  @param {!HTMLVideoElement} video The video element.
+ * @export
+ */
+shaka.StatsOverlay.prototype.init = function(player, canvas, video) {
+
+  this.player_ = player;
+  this.canvas_ = canvas;
+  this.video_ = video;
+
+  this.width_ = canvas.width;
+  this.height_ = canvas.height - 5;
```

Comment:
The intention was to add some padding to the bottom, for a time series, but I haven't taken advantage of that yet.